pkg/scheduler/schedule_one.go

```diff
@@ -89,10 +90,43 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 	podsToActivate := framework.NewPodsToActivate()
 	state.Write(framework.PodsToActivateKey, podsToActivate)
 
-	schedulingCycleCtx, cancel := context.WithCancel(ctx)
-	defer cancel()
-	scheduleResult, assumedPodInfo := sched.schedulingCycle(schedulingCycleCtx, state, fwk, podInfo, podsToActivate, start)
-	if scheduleResult.FeasibleNodes == 0 {
+	var scheduleResult ScheduleResult
+	var assumedPodInfo *framework.QueuedPodInfo
+
+	func() {
+		schedulingCycleCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+		scheduleResult, assumedPodInfo, err = sched.schedulingCycle(schedulingCycleCtx, state, fwk, podInfo, start)
+		if err != nil {
```

Comment:
why not make this a function that is called inside `schedulingCycle`

Then we guarantee that everything is using `schedulingCycleCtx` without having to type it multiple times.

Reply:
I think the challenge of putting everything inside schedulingCycle is the excessive amount of return paths, which may not be neater than the current style.

Reply:
I think it's a traditionally golang style, if we have error, let's handle them outside of the original function.

> Then we guarantee that everything is using schedulingCycleCtx without having to type it multiple times.

We can mitigate this by changing to `ctx, cancel := context.WithCancel(ctx)`, but `schedulingCycle` seems explicit to me.